test/metric_learn_test.py

```diff
@@ -76,6 +77,124 @@ def test_singular_returns_pseudo_inverse(self):
                     pseudo_inverse)
 
 
+class TestSCML(MetricTestCase):
+  def test_iris(self):
+    scml = SCML_Supervised()
+    scml.fit(self.iris_points, self.iris_labels)
+
+    csep = class_separation(scml.transform(self.iris_points), self.iris_labels)
+    self.assertLess(csep, 0.3)
+
+  def test_bad_basis(self):
+    scml = SCML(basis='bad_basis')
+    triplets = np.ones((3, 3, 3))
+    authorized_basis = ['triplet_diffs']
+    msg = ("`basis` must be one of the options '{}' or an array of shape "
+           "(n_basis, n_features).".format("', '".join(authorized_basis)))
+    with pytest.raises(ValueError) as raised_error:
+      scml.fit(triplets)
+    assert msg == raised_error.value.args[0]
+
+  def test_big_n_basis(self):
+    scml = SCML(n_basis=4)
+    triplets = np.random.rand(3, 3, 3)
+    n_basis = 3
+    msg = ("The selected number of basis is greater than the number of points"
+           ", only n_basis = %d will be generated" % n_basis)
+    with pytest.warns(UserWarning) as raised_warning:
+      scml.fit(triplets)
+    assert msg == raised_warning[0].message.args[0]
+
+  def test_n_basis_wrong_type(self):
+    X = np.array([[0, 0], [0, 1], [2, 0], [2, 1]])
+    y = np.array([1, 0, 1, 0])
+
+    constraints = Constraints(y)
+    triplets = constraints.generate_knntriplets(X, k_genuine=1, k_impostor=1)
+    triplets = X[triplets]
+
+    n_basis = 4.0
+
+    scml = SCML(n_basis=n_basis)
+    msg = ("n_basis should be an integer, instead it is of type %s"
+           % type(n_basis))
+    with pytest.raises(ValueError) as raised_error:
+      scml.fit(triplets)
+    assert msg == raised_error.value.args[0]
+
+  def test_bad_basis_supervised(self):
```

Comment:
Ok! Tried to parametrize it but without any success, it looks like it's not possible to parametrize inside a `unittest.TestCase` class! [https://stackoverflow.com/questions/35561843/py-test-parametrize-test-cases-from-classes](url)

Reply:
True. This is why some ITML tests (e.g., `test_bounds_parameters_invalid`) are outside the class. You could do the same here.

IMO, ultimately we should remove this class structure (which we inherited from the initial use of `unittest` before we moved mostly to `pytest`) or make it a subclass of `object`as indicated in the stackoverflow topic you mentioned

Reply:
Making the SCML test inherit from `object` would be a pretty straightforward solution as there is only one assert from `unittest` and the rest are normal asserts. 

Maybe for the other classes, it may require a little bit more work. But I agree that it makes sense to move in that direction!

Reply:
Another alternative would be to make `MetricTestCase` inherit from object and build the required assertion methods that are used.

Reply:
You should go for the easiest fix, and avoid making too many changes unrelated with this PR

But of course if you prefer you are very welcome to first remove the last remaining bits of `unittest` in a separate PR ;-)